### PR TITLE
Implemented few operations for the wasm instruction set

### PR DESCRIPTION
**Pull request by anutosh491**
The following operations would be implemented through the PR

- [x] reduce_max
- [x] reduce_min
- [x] swizzle
- [x] bitwise_cast
- [x] shuffle
- [x] fast_cast

### CODE REVIEW

File: include/xsimd/arch/xsimd_wasm.hpp [hunk 1]
```diff
@@ -415,6 +427,59 @@ namespace xsimd
...
+        {
+            template <class A>
+            inline batch<float, A> fast_cast(batch<int32_t, A> const& self, batch<float, A> const&, requires_arch<wasm>) noexcept
+            {
+                return wasm_f32x4_convert_i32x4(self);
+            }
+
+            template <class A>
+            inline batch<float, A> fast_cast(batch<uint32_t, A> const& v, batch<float, A> const&, requires_arch<wasm>) noexcept
+            {
```
Comment (serge-sans-paille) [inline include/xsimd/arch/xsimd_wasm.hpp:441]: This is exactly the same operation we use for sse2... I think we should make it generic, I'll give it a try.

File: include/xsimd/arch/xsimd_wasm.hpp [hunk 2]
```diff
@@ -1039,6 +1104,44 @@ namespace xsimd
             return wasm_f64x2_extract_lane(tmp2, 0);
         }
 
+        // reduce_max
+        template <class A, class T, class _ = typename std::enable_if<(sizeof(T) <= 2), void>::type>
```
Comment (serge-sans-paille) [inline include/xsimd/arch/xsimd_wasm.hpp:1108]: Isn't this exactly the same code as the one we have for the generic implementation?

File: include/xsimd/arch/xsimd_wasm.hpp [hunk 3]
```diff
@@ -415,6 +443,72 @@ namespace xsimd
...
+            inline batch<int32_t, A> fast_cast(batch<float, A> const& self, batch<int32_t, A> const&, requires_arch<wasm>) noexcept
+            {
+                return wasm_i32x4_make(
+                    static_cast<int32_t>(wasm_f32x4_extract_lane(self, 0)),
+                    static_cast<int32_t>(wasm_f32x4_extract_lane(self, 1)),
+                    static_cast<int32_t>(wasm_f32x4_extract_lane(self, 2)),
+                    static_cast<int32_t>(wasm_f32x4_extract_lane(self, 3)));
+            }
+
+            template <class A>
```
Comment (serge-sans-paille) [inline include/xsimd/arch/xsimd_wasm.hpp:492]: This one should be redundant once #964 is merged

Comment (anutosh491) [reply include/xsimd/arch/xsimd_wasm.hpp:492]: Yes.

Comment (anutosh491) [reply include/xsimd/arch/xsimd_wasm.hpp]: Addressed

**Comment by anutosh491** [comment]
cc @JohanMabille not sure if the failing check is related to the changes on the branch. I think they might not be related :(

**Comment by JohanMabille** [comment]
The failure is indeed unrelated, I need to update the SDE version.

**Comment by JohanMabille** [comment]
@anutosh491 I've fixed the issue on master, can you rebase your branch to check everything is fine?

**Pull request merged**